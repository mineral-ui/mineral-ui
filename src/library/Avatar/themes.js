/* @flow */
import type { AvatarThemeFn } from './types';

export const avatarTheme: AvatarThemeFn = (baseTheme) => ({
  Avatar_fontSize_small: baseTheme.fontSize_mouse,
  Avatar_fontSize_medium: baseTheme.fontSize_ui,
  Avatar_fontSize_large: baseTheme.h4_fontSize,
  Avatar_fontSize_jumbo: baseTheme.h4_fontSize,
  Avatar_fontWeight: baseTheme.fontWeight_bold,
  Avatar_size_small: baseTheme.size_small,
  Avatar_size_medium: baseTheme.size_medium,
  Avatar_size_large: baseTheme.size_large,
  Avatar_size_jumbo: baseTheme.size_jumbo,

  ...baseTheme
});
