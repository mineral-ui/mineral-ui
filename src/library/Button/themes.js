/* @flow */
import { pxToEm } from '../styles';

import type { ButtonThemeFn } from './types';

// prettier-ignore
export const buttonTheme: ButtonThemeFn = (baseTheme) => ({
  Button_backgroundColor: baseTheme.backgroundColor,
  Button_backgroundColor_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_focus: baseTheme.backgroundColor_focus,
  Button_backgroundColor_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_minimal_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_minimal_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_primary: baseTheme.backgroundColor_themePrimary,
  Button_backgroundColor_primary_active: baseTheme.backgroundColor_themePrimary_active,
  Button_backgroundColor_primary_focus: baseTheme.backgroundColor_themePrimary_focus,
  Button_backgroundColor_primary_hover: baseTheme.backgroundColor_themePrimary_hover,
  Button_borderColor: baseTheme.borderColor,
  Button_borderColor_active: baseTheme.borderColor_theme_active,
  Button_borderColor_focus: baseTheme.borderColor_theme_focus,
  Button_borderColor_hover: baseTheme.borderColor_theme_hover,
  Button_borderRadius: baseTheme.borderRadius_3,
  Button_borderWidth: 1, // px,
  Button_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${baseTheme.borderColor_theme_focus}`,
  Button_color: baseTheme.color_theme,
  Button_color_minimal: baseTheme.color_theme,
  Button_color_primary: baseTheme.color_themePrimary,
  Button_fontWeight: baseTheme.fontWeight_semiBold,
  Button_paddingHorizontal: baseTheme.space_inset_sm_plus,
  Button_paddingIconOnly_small: pxToEm(3),
  Button_paddingIconOnly_medium: pxToEm(7),
  Button_paddingIconOnly_large: pxToEm(7),
  Button_paddingIconOnly_jumbo: pxToEm(13),
  Button_height_small: baseTheme.size_small,
  Button_height_medium: baseTheme.button_size_medium,
  Button_height_large: baseTheme.button_size_large,
  Button_height_jumbo: baseTheme.button_size_jumbo,

  ButtonContent_fontSize: baseTheme.fontSize_ui_sm,
  ButtonContent_fontSize_small: pxToEm(12),

  ButtonIcon_color: baseTheme.icon_color_theme,
  ButtonIcon_margin: baseTheme.space_inset_sm,

  ...baseTheme
});
