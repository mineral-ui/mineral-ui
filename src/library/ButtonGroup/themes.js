/* @flow */
import type { ButtonGroupThemeFn } from './types';

export const buttonGroupTheme: ButtonGroupThemeFn = (baseTheme) => ({
  ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
  ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
  ButtonGroupButton_borderColor_active: baseTheme.borderColor_theme_active,
  ButtonGroupButton_borderColor_hover: baseTheme.borderColor_theme_hover,
  ButtonGroupButton_borderStartColor: baseTheme.borderColor,
  ButtonGroupButton_borderStartColor_checked: 'currentcolor',
  ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,

  ...baseTheme
});
