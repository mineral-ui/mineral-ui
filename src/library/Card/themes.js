/* @flow */
import { pxToEm } from '../styles';

import type {
  CardThemeFn,
  CardActionsThemeFn,
  CardBlockThemeFn,
  CardDividerThemeFn,
  CardFooterThemeFn,
  CardStatusThemeFn,
  CardTitleThemeFn
} from './types';

// prettier-ignore
export const cardTheme: CardThemeFn = (baseTheme) => ({
  Card_backgroundColor: baseTheme.panel_backgroundColor,
  Card_borderColor: baseTheme.panel_borderColor,
  Card_borderRadius: baseTheme.borderRadius_1,
  Card_boxShadow: baseTheme.boxShadow_2,
  Card_boxShadow_focus: `0 0 0 2px ${baseTheme.borderColor_theme_focus}, ${baseTheme.boxShadow_1}`,

  CardRow_marginVertical: baseTheme.space_stack_md,
  CardRow_marginVerticalLast: baseTheme.space_stack_lg,
  CardRow_paddingHorizontal: baseTheme.space_inset_md,

  ...baseTheme
});

export const cardActionsTheme: CardActionsThemeFn = (baseTheme) => ({
  CardActionsAction_spaceInline: baseTheme.space_inline_sm,

  ...baseTheme
});

export const cardBlockTheme: CardBlockThemeFn = (baseTheme) => ({
  CardBlock_fontSize: baseTheme.fontSize_ui,
  CardBlock_lineHeight: baseTheme.lineHeight_prose,

  ...baseTheme
});

export const cardDividerTheme: CardDividerThemeFn = (baseTheme) => ({
  CardDivider_borderColor: baseTheme.borderColor,
  CardDivider_borderWidth: '1px',

  ...baseTheme
});

export const cardFooterTheme: CardFooterThemeFn = (baseTheme) => ({
  CardFooter_backgroundColor: baseTheme.well_backgroundColor,
  CardFooter_borderColor: baseTheme.borderColor,

  CardFooterRow_marginVertical: baseTheme.space_stack_sm,
  CardFooterRow_marginVerticalLast: baseTheme.space_stack_md,

  CardFooterTitle_color: baseTheme.h5_color,
  CardFooterTitle_fontSize: baseTheme.h5_fontSize,
  CardFooterTitle_fontWeight: baseTheme.h5_fontWeight,

  ...baseTheme
});

export const cardStatusTheme: CardStatusThemeFn = (baseTheme) => ({
  CardStatus_fontSize: baseTheme.fontSize_ui,
  CardStatus_fontWeight: baseTheme.fontWeight_regular,

  CardStatusIcon_margin: baseTheme.space_inline_sm,
  CardStatusIcon_size: pxToEm(12),

  ...baseTheme
});

export const cardTitleTheme: CardTitleThemeFn = (baseTheme) => ({
  CardTitle_color: baseTheme.h4_color,
  CardTitle_fontSize: baseTheme.h4_fontSize,
  CardTitle_fontWeight: baseTheme.h4_fontWeight,

  CardTitleAvatar_margin: baseTheme.space_inline_sm,
  CardTitleAvatarSize: baseTheme.size_small,
  CardTitleAvatarSize_large: baseTheme.size_large,

  CardTitleIcon_margin: baseTheme.space_inline_sm,

  CardTitleSecondaryText_color: baseTheme.color_mouse,
  CardTitleSecondaryText_fontSize: baseTheme.fontSize_mouse,
  CardTitleSecondaryText_fontWeight: baseTheme.fontWeight_regular,

  CardSubtitle_color: baseTheme.color_mouse,
  CardSubtitle_fontSize: baseTheme.fontSize_mouse,
  CardSubtitle_fontWeight: baseTheme.fontWeight_regular,
  CardSubtitle_marginTop: baseTheme.space_stack_sm,

  ...baseTheme
});
