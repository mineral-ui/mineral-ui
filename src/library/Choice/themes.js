/* @flow */
import { pxToEm } from '../styles';

import type { ChoiceThemeFn, ChoiceGroupThemeFn } from './types';

export const choiceTheme: ChoiceThemeFn = (baseTheme) => {
  const colors = {
    background: baseTheme.input_backgroundColor,
    regular: baseTheme.borderColor,
    checked: baseTheme.borderColor_theme,
    focus: baseTheme.borderColor_theme_focus,
    hover: baseTheme.borderColor_theme_hover
  };

  // prettier-ignore
  return {
    ChoiceControl_backgroundColor: colors.background,
    ChoiceControl_backgroundColor_checked: colors.checked,
    ChoiceControl_backgroundColor_checkedHover: colors.hover,
    ChoiceControl_borderColor: colors.regular,
    ChoiceControl_borderColor_hover: colors.hover,
    ChoiceControl_borderColor_checked: colors.checked,
    ChoiceControl_borderColor_checkedHover: colors.hover,
    ChoiceControl_borderRadius: baseTheme.borderRadius_1,
    ChoiceControl_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${colors.focus}`,
    ChoiceControl_size: pxToEm(16),
    ChoiceControl_size_jumbo: pxToEm(24),

    ChoiceText_color: baseTheme.color,
    ChoiceText_fontSize: baseTheme.fontSize_ui,
    ChoiceText_fontSize_small: pxToEm(12),
    ChoiceText_marginHorizontal: baseTheme.space_inline_md,

    ...baseTheme
  };
};

export const choiceGroupTheme: ChoiceGroupThemeFn = (baseTheme) => ({
  ChoiceGroupControl_marginHorizontal_inline: baseTheme.space_inline_xl,
  ChoiceGroupControl_marginVertical_stacked: baseTheme.space_stack_md,
  ChoiceGroupControl_marginVertical_stackedJumbo: baseTheme.space_stack_lg,

  ...baseTheme
});
