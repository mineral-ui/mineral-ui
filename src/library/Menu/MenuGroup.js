/* @flow */
import React from 'react';
import { createStyledComponent } from '../styles';
import MenuGroupTitle from './MenuGroupTitle';

type Props = {
  /** [MenuItems](/components/menu-item) */
  children?: React$Node,
  /** Title for the group */
  title?: React$Node
};

export const componentTheme = (baseTheme: Object) => ({
  MenuGroup_margin: baseTheme.space_stack_sm,

  ...baseTheme
});

const Root = createStyledComponent(
  'div',
  props => {
    const theme = componentTheme(props.theme);

    return {
      margin: `${theme.MenuGroup_margin} 0`,
      '&:first-child,& + &': {
        marginTop: 0
      },

      '&:last-child': {
        marginBottom: 0
      }
    };
  },
  {
    displayName: 'MenuGroup'
  }
);

/**
 * MenuGroups assemble [MenuItems](/components/menu-item) and can display a title.
 * Grouping Menu options provides context clues to users about related actions.
 * An optional title can be added to reinforce the intent of the grouping.
 */
export default function MenuGroup({ children, title, ...restProps }: Props) {
  return (
    <Root {...restProps}>
      {title && <MenuGroupTitle>{title}</MenuGroupTitle>}
      {children}
    </Root>
  );
}
