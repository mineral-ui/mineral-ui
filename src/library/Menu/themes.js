/* @flow */
import type {
  MenuDividerThemeFn,
  MenuGroupThemeFn,
  MenuGroupTitleThemeFn,
  MenuItemThemeFn
} from './types';

export const menuDividerTheme: MenuDividerThemeFn = (baseTheme) => ({
  MenuDivider_borderColor: baseTheme.borderColor,
  MenuDivider_borderWidth: '1px',
  MenuDivider_margin: baseTheme.space_stack_sm,

  ...baseTheme
});

export const menuGroupTheme: MenuGroupThemeFn = (baseTheme) => ({
  MenuGroup_margin: baseTheme.space_stack_sm,

  ...baseTheme
});

// [1] Deviation from h5 for optics relative to a MenuItem
export const menuGroupTitleTheme: MenuGroupTitleThemeFn = (baseTheme) => ({
  MenuGroupTitle_fontSize: baseTheme.fontSize_mouse, // [1]
  MenuGroupTitle_fontWeight: baseTheme.h5_fontWeight,
  MenuGroupTitle_paddingTop: baseTheme.space_stack_md,
  MenuGroupTitle_paddingBottom: baseTheme.space_stack_sm,

  ...baseTheme
});

// Some of these values (all of the margins & paddings and the content fontSize)
// come from Button (large)
// prettier-ignore
export const menuItemTheme: MenuItemThemeFn = (baseTheme) => ({
  MenuItem_backgroundColor_selected: baseTheme.backgroundColor_theme_selected,
  MenuItem_backgroundColor_active: baseTheme.color_gray_20,
  MenuItem_backgroundColor_focus: baseTheme.color_gray_10,
  MenuItem_backgroundColor_hover: baseTheme.color_gray_10,
  MenuItem_backgroundColor_selectedActive: baseTheme.backgroundColor_theme_selectedActive,
  MenuItem_backgroundColor_selectedHover: baseTheme.backgroundColor_theme_selectedHover,
  MenuItem_color: baseTheme.color,
  MenuItem_fontWeight: baseTheme.fontWeight_regular,
  MenuItem_fontWeight_selected: baseTheme.fontWeight_bold,
  MenuItem_paddingHorizontal: baseTheme.space_inset_md,
  MenuItem_paddingVertical: baseTheme.space_inset_sm,

  MenuItemContent_fontSize: baseTheme.fontSize_ui,

  MenuItemIcon_color: baseTheme.icon_color_theme,
  MenuItemIcon_margin: baseTheme.space_inset_sm,

  MenuItemSecondaryText_color: baseTheme.color_mouse,
  MenuItemSecondaryText_fontSize: baseTheme.fontSize_mouse,

  ...baseTheme
});
