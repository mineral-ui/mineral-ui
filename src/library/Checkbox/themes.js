/* @flow */
import { mapComponentThemes } from '../themes';
import { choiceTheme, choiceGroupTheme } from '../Choice';

import type { CheckboxThemeFn, CheckboxGroupThemeFn } from './types';

import { pxToEm } from '../styles';

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxTheme: CheckboxThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'Choice',
      theme: choiceTheme(baseTheme)
    },
    {
      name: 'Checkbox',
      theme: {
        CheckboxControl_backgroundColor_checked: baseTheme.checkbox_backgroundColor_checked,
        CheckboxControl_backgroundColor_checkedHover: baseTheme.checkbox_backgroundColor_checkedHover,
        CheckboxControl_borderColor: baseTheme.checkbox_borderColor,
        CheckboxControl_borderColor_checked: baseTheme.checkbox_borderColor_checked,
        CheckboxControl_borderColor_checkedHover: baseTheme.checkbox_borderColor_checkedHover,
        CheckboxControl_borderColor_hover: baseTheme.checkbox_borderColor_hover,
        CheckboxControl_borderRadius: baseTheme.borderRadius_2,
        CheckboxControl_borderWidth: `${baseTheme.checkbox_borderWidth}px`,
        CheckboxControl_size: pxToEm(20),
        CheckboxText_marginHorizontal: baseTheme.space_inline_sm
      }
    },
    baseTheme
  );

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxGroupTheme: CheckboxGroupThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'ChoiceGroup',
      theme: choiceGroupTheme(baseTheme)
    },
    {
      name: 'CheckboxGroup',
      theme: {}
    },
    baseTheme
  );
