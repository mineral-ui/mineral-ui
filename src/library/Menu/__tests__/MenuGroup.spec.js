/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import MenuGroup, { componentTheme } from '../MenuGroup';
import examples from '../../../website/app/demos/Menu/examples/MenuGroup';
import testDemoExamples from '../../../../utils/testDemoExamples';
import testThemeOverrides from '../../../../utils/testThemeOverrides';
import { getProcessedComponentThemeKeys } from '../../themes/processComponentTheme';

function shallowMenuGroup(props = {}) {
  const menuGroupProps = {
    ...props
  };

  return shallow(<MenuGroup {...menuGroupProps} />);
}

describe('MenuGroup', () => {
  testDemoExamples(examples);

  describe('renders', () => {
    it('root', () => {
      const menuGroup = shallowMenuGroup();

      expect(menuGroup.exists()).toEqual(true);
    });
  });

  describe('theme overrides', () => {
    testThemeOverrides(
      <MenuGroup />,
      getProcessedComponentThemeKeys(componentTheme)
    );
  });
});
