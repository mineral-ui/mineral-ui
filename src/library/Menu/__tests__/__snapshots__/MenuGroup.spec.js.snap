// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuGroup demo examples menu-group 1`] = `
.css-0,
[data-css-0] {
  overflow: hidden;
}

.css-0 > div,
[data-css-0] > div {
  background-color: #fff;
  float: left;
  margin: 0 0.5rem 0.5rem 0;
  width: 14em;
}

.css-1,
[data-css-1] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
}

.css-1 *,
[data-css-1] *,
.css-1 *::before,
[data-css-1] *::before,
.css-1 *::after,
[data-css-1] *::after {
  box-sizing: inherit;
}

.css-2,
[data-css-2] {
  margin: 0.5em 0;
}

.css-2:first-child,
[data-css-2]:first-child,
.css-2 + .css-2,
[data-css-2] + [data-css-2] {
  margin-top: 0;
}

.css-2:last-child,
[data-css-2]:last-child,
.css-2[data-simulate-lastchild],
[data-css-2][data-simulate-lastchild] {
  margin-bottom: 0;
}

.css-8,
[data-css-8] {
  font-size: 0.75em;
  font-weight: 700;
  margin: 0;
  padding: 1.3333333333333333em 1.3333333333333333em 0.6666666666666666em;
}

.css-9,
[data-css-9] {
  color: #333840;
  cursor: pointer;
  font-weight: 400;
  padding: 0.5em 1em;
}

.css-9:focus,
[data-css-9]:focus,
.css-9[data-simulate-focus],
[data-css-9][data-simulate-focus] {
  background-color: #f5f7fa;
  outline: 0;
}

.css-9:hover,
[data-css-9]:hover,
.css-9[data-simulate-hover],
[data-css-9][data-simulate-hover] {
  background-color: #f5f7fa;
}

.css-9:active,
[data-css-9]:active,
.css-9[data-simulate-active],
[data-css-9][data-simulate-active] {
  background-color: #ebeff5;
}

.css-9 [role="img"],
[data-css-9] [role="img"] {
  box-sizing: content-box;
  display: block;
  fill: #2e6fd9;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

.css-9 [role="img"]:first-child,
[data-css-9] [role="img"]:first-child {
  margin-right: 0.5em;
}

.css-9 [role="img"]:last-child,
[data-css-9] [role="img"]:last-child {
  margin-left: 0.5em;
}

.css-20,
[data-css-20] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: space-between;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.css-21,
[data-css-21] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex: 1 1 auto;
  flex-wrap: wrap;
  font-size: 0.875em;
  justify-content: space-between;
  padding: 0.21428571428571427em 0 0.2857142857142857em;
  white-space: normal;
  word-break: break-all;
  -webkit-flex: 1 1 auto;
  -webkit-box-lines: multiple;
  -webkit-flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.css-22,
[data-css-22] {
  margin-right: 0.5714285714285714em;
  word-break: break-word;
}

.css-23,
[data-css-23] {
  color: #58606e;
  font-size: 0.75em;
  padding-top: 0.25em;
  word-break: break-word;
}

<glamorous(div)>
  <div
    className="css-0"
  >
    <Menu>
      <Menu>
        <div
          className="css-1"
        >
          <MenuGroup
            title="Menu Group Title"
          >
            <MenuGroup>
              <div
                className="css-2"
              >
                <MenuGroupTitle>
                  <MenuGroupTitle>
                    <h3
                      className="css-8"
                    >
                      Menu Group Title
                    </h3>
                  </MenuGroupTitle>
                </MenuGroupTitle>
                <MenuItem>
                  <MenuItem
                    tabIndex={0}
                  >
                    <div
                      className="css-9"
                      tabIndex={0}
                    >
                      <glamorous(span)>
                        <span
                          className="css-20"
                        >
                          <glamorous(span)>
                            <span
                              className="css-21"
                            >
                              <glamorous(span)>
                                <span
                                  className="css-22"
                                >
                                  Menu item
                                </span>
                              </glamorous(span)>
                              <glamorous(span)>
                                <span
                                  className="css-23"
                                />
                              </glamorous(span)>
                            </span>
                          </glamorous(span)>
                        </span>
                      </glamorous(span)>
                    </div>
                  </MenuItem>
                </MenuItem>
                <MenuItem>
                  <MenuItem
                    tabIndex={0}
                  >
                    <div
                      className="css-9"
                      tabIndex={0}
                    >
                      <glamorous(span)>
                        <span
                          className="css-20"
                        >
                          <glamorous(span)>
                            <span
                              className="css-21"
                            >
                              <glamorous(span)>
                                <span
                                  className="css-22"
                                >
                                  Menu item
                                </span>
                              </glamorous(span)>
                              <glamorous(span)>
                                <span
                                  className="css-23"
                                />
                              </glamorous(span)>
                            </span>
                          </glamorous(span)>
                        </span>
                      </glamorous(span)>
                    </div>
                  </MenuItem>
                </MenuItem>
              </div>
            </MenuGroup>
          </MenuGroup>
          <MenuGroup
            title="Menu Group Title"
          >
            <MenuGroup>
              <div
                className="css-2"
              >
                <MenuGroupTitle>
                  <MenuGroupTitle>
                    <h3
                      className="css-8"
                    >
                      Menu Group Title
                    </h3>
                  </MenuGroupTitle>
                </MenuGroupTitle>
                <MenuItem>
                  <MenuItem
                    tabIndex={0}
                  >
                    <div
                      className="css-9"
                      tabIndex={0}
                    >
                      <glamorous(span)>
                        <span
                          className="css-20"
                        >
                          <glamorous(span)>
                            <span
                              className="css-21"
                            >
                              <glamorous(span)>
                                <span
                                  className="css-22"
                                >
                                  Menu item
                                </span>
                              </glamorous(span)>
                              <glamorous(span)>
                                <span
                                  className="css-23"
                                />
                              </glamorous(span)>
                            </span>
                          </glamorous(span)>
                        </span>
                      </glamorous(span)>
                    </div>
                  </MenuItem>
                </MenuItem>
                <MenuItem>
                  <MenuItem
                    tabIndex={0}
                  >
                    <div
                      className="css-9"
                      tabIndex={0}
                    >
                      <glamorous(span)>
                        <span
                          className="css-20"
                        >
                          <glamorous(span)>
                            <span
                              className="css-21"
                            >
                              <glamorous(span)>
                                <span
                                  className="css-22"
                                >
                                  Menu item
                                </span>
                              </glamorous(span)>
                              <glamorous(span)>
                                <span
                                  className="css-23"
                                />
                              </glamorous(span)>
                            </span>
                          </glamorous(span)>
                        </span>
                      </glamorous(span)>
                    </div>
                  </MenuItem>
                </MenuItem>
              </div>
            </MenuGroup>
          </MenuGroup>
        </div>
      </Menu>
    </Menu>
  </div>
</glamorous(div)>
`;
